### PR TITLE
Added testcases for subcommand testing

### PR DESCRIPTION
**Pull request by Rithesh17**
Added test cases to validate the subcommand execution.

### CODE REVIEW

File: src/subcommand.go [hunk 1]
```diff
@@ -15,6 +15,20 @@ func printCmdErr(e string) {
 	fmt.Println(fmt.Sprintf("%s\n Try --help to list subcommands and options.", e))
 }
 
+func reorderArgs(args []string) []string {
```
Comment (fredxfred) [inline src/subcommand.go]: Why did you add this? Is it necessary?

Comment (Rithesh17) [reply src/subcommand.go]: Removed it

File: src/tests/subcommand_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	// Assertion for command output
+	if !strings.Contains(output, assertStatement) {
+		return fmt.Errorf("Expected output to contain %s, but got: %s", assertStatement, output)
+	}
+
+	return nil
+}
+
+func TestCommandRun(t *testing.T) {
```
Comment (fredxfred) [inline src/tests/subcommand_test.go:34]: Please run setup.sh from within here so that our unit tests remain hermetic and runnable directly after cloning the repo

File: src/tests/subcommand_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+// runAndAssertSubcommand executes a subcommand of the "semantifly" command with the provided arguments
+// and asserts that the output contains the specified assertion statement.
+//
+// Parameters:
+//   - subcommand (string): The subcommand to execute.
+//   - assertStatement (string): The expected substring to be present in the command output.
+//   - args ([]string): Additional arguments to pass to the subcommand.
+func runAndAssertSubcommand(subcommand string, assertStatement string, args []string) error {
```
Comment (fredxfred) [inline src/tests/subcommand_test.go]: nit: assertStatement -> wantedStdoutSubstr

We are not actually directly asserting that arg, and "got"/"want" are golang naming conventions for unit test assertion args; this is a "want" variable

Comment (fredxfred) [inline src/tests/subcommand_test.go]: I feel like

```
tempFile, err := os.CreateTemp("", "semantifly_test_*.txt")
...
args := []string{tempFile.Name()}
runAndAssertSubcommand("add", "Added file successfully", args); err != nil {
```

is overly complicated and runAndAssertSubcommand is kind of convoluted/overengineered. I would prefer something like ```runAndCheckStdoutContains("semantifly add semantifly_test.txt", "Added file successfully")``` because it's more readable and obvious what it does.

File: src/tests/subcommand_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	if err := runAndAssertSubcommand("add", "Added file successfully", args); err != nil {
+		t.Errorf("Failed to execute 'add' subcommand: %v", err)
+	}
+
+	// Testing Get subcommand
+	if err := runAndAssertSubcommand("get", testContent, args); err != nil {
+		t.Errorf("Failed to execute 'add' subcommand: %v", err)
+	}
+
+	// Testing Delete subcommand
```
Comment (fredxfred) [inline src/tests/subcommand_test.go:85]: Please add non-happy-path test cases in either this unit test or another one

File: src/tests/subcommand_test.go (hunk 3 above)
Comment (Rithesh17) [reply src/tests/subcommand_test.go]: I have added the args section to later add the flags for testing the subcommands. In which case we just add to the args list and call the function instead of writing a new command string.

Comment (Rithesh17) [reply src/tests/subcommand_test.go]: Got it

File: src/tests/subcommand_test.go (hunk 2 above)
Comment (Rithesh17) [reply src/tests/subcommand_test.go:34]: Added

File: src/tests/subcommand_test.go (hunk 3 above)
Comment (fredxfred) [reply src/tests/subcommand_test.go]: Ahh I think I see now, it is weird that the golang exec.Command requires args to be formatted like that.

File: src/tests/subcommand_test.go (hunk 4 above)
Comment (Rithesh17) [reply src/tests/subcommand_test.go:85]: Added

File: src/tests/subcommand_test.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	if err := runAndCheckStdoutContains("get", "not found in the index", args); err != nil {
+		t.Errorf("Failed to execute 'get' subcommand: %v", err)
+	}
+
+	// Testing Delete subcommand after deleting the entry
+	if err := runAndCheckStdoutContains("delete", "Entry not found in index", args); err != nil {
+		t.Errorf("Failed to execute 'delete' subcommand: %v", err)
+	}
+
+	defer os.Remove("index.list")
```
Comment (fredxfred) [inline src/tests/subcommand_test.go]: nit: move this up to right after the initial add command succeeds

**Review by fredxfred — APPROVED** [review]

File: src/tests/subcommand_test.go (hunk 5 above)
Comment (Rithesh17) [reply src/tests/subcommand_test.go]: Done
